lib/schema_plus/compatibility/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -3,9 +3,19 @@ module ActiveRecord
     module ConnectionAdapters
       module AbstractAdapter
         def tables_without_deprecation
-          ActiveSupport::Deprecation.silence do
-            tables
-          end
+          # In the future (AR 5.1?) tables may really return just tables instead of data
+          # sources, and the deprecation warning will be removed. We would
+          # have to update the method then.
+          #
+          # For consistency's sake, this method is overridden in all standard adapters
+          # (SQLite3, MySQL and Postgres) with a proper tables-only implementation, so you're
+          # really going to get ONLY tables and no warning for these 3 databases across all
+          # current and future versions of AR.
+          # For non-supported DBs, try our best by trying first to subtract #views from #data_sources
+          # but we have to return the plain old #tables result in AR4.2.
+          all_sources = (try :data_sources) || tables
+          views_only = (try :views) || []
+          all_sources - views_only
         end
```

Comment:
I think it's safe to say we don't support non-supported DBs :)  

So I'd remove this method entirely, rather than keeping around code that's not tested (and likely never used).